docs/reference/api/connectivity/bluetooth/Gap.md

```diff
@@ -1,11 +1,19 @@
 ## Gap
 
-[Add description here.]
+Generic Access Profile is the layer of the stack that handles connectivity tasks. This includes link establishment and termination, advertising and scanning.
+
+Devices with data to publish can use GAP to advertise. They can either include the data in the advertisement itself, inside the scan response, or leave it to be queried by a peer device after connection has been established.
+
+The other side of the process is the act of scanning which listens for advertisements, allows us to query the advertisers for more data via a scan request or to connect in order to query the GATT server on the peer device for the data we want.
```

Comment:
I'm not sure that it is necessary to mention GATT server. A master can act as a server and wait for the slave to query its server. As an example that's how Apple's ANCS works.